march_hardware/test/TestEncoder.cpp

```diff
@@ -29,54 +41,67 @@ class TestEncoderParameterizedValidTarget : public TestEncoder,
+  ASSERT_EQ(this->encoder.getUpperSoftLimitIU(), this->upper_soft_limit);
 }
 
-TEST(TestEncoderConstructor, ResolutionAboveRange)
+TEST_F(TestEncoder, ResolutionBelowRange)
 {
-  ASSERT_THROW(march::Encoder(50, 0, 0, 0, 0, 0, 0), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(0, this->lower_limit, this->upper_limit, this->lower_limit_rad, this->upper_limit_rad,
+                              this->lower_soft_limit_rad, this->upper_soft_limit_rad),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, DifferentIuPerRad)
+TEST_F(TestEncoder, ResolutionAboveRange)
 {
-  ASSERT_THROW(march::Encoder(12, 2, 1, 1.0, 2.0, 1.0, 2.0), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(50, this->lower_limit, this->upper_limit, this->lower_limit_rad, this->upper_limit_rad,
+                              this->lower_soft_limit_rad, this->upper_soft_limit_rad),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, InvalidSoftLimits)
+TEST_F(TestEncoder, LowerSoftLimitAboveUpperSoftLimit)
 {
-  ASSERT_THROW(march::Encoder(17, 2053, 45617, -0.34906585, 1.745329252, 0.4, 0.1), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(this->resolution, this->lower_limit, this->upper_limit, this->lower_limit_rad,
+                              this->upper_limit_rad, 0.4, 0.1),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, InvalidLowerSoftLimit)
+TEST_F(TestEncoder, LowerSoftLimitLowerThanLowerHardLimit)
 {
-  ASSERT_THROW(march::Encoder(17, 2053, 45617, -0.34906585, 1.745329252, -0.4, 1), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(this->resolution, this->lower_limit, this->upper_limit, this->lower_limit_rad,
+                              this->upper_limit_rad, -0.4, 1),
+               march::error::HardwareException);
 }
 
-TEST(TestEncoderConstructor, InvalidUpperSoftLimit)
+TEST_F(TestEncoder, UpperSoftLimitHigherThanUpperHardLimit)
 {
-  ASSERT_THROW(march::Encoder(17, 2053, 45617, -0.34906585, 1.745329252, -0.3, 2.0), march::error::HardwareException);
+  ASSERT_THROW(march::Encoder(this->resolution, this->lower_limit, this->upper_limit, this->lower_limit_rad,
+                              this->upper_limit_rad, -0.3, 2.0),
+               march::error::HardwareException);
 }
 
 INSTANTIATE_TEST_CASE_P(ParameterizedLimits, TestEncoderParameterizedLimits,
-                        testing::Values(std::make_tuple(2052, false), std::make_tuple(2053, false),
-                                        std::make_tuple(2054, true), std::make_tuple(45616, true),
-                                        std::make_tuple(45617, false), std::make_tuple(45618, false),
+                        testing::Values(std::make_tuple(2053 - 1, false), std::make_tuple(2053, false),
+                                        std::make_tuple(2053 + 1, true), std::make_tuple(45617 - 1, true),
```

Comment:
```suggestion
                                        std::make_tuple(this->lower_hard_limit + 1, true), std::make_tuple(this->upper_hard_limit - 1, true),
```
Also for line above and below